app/controllers/users/folders_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Users::FoldersController < ApplicationController
+
+  def show
+    current_url = request.env["PATH_INFO"]
+    username = current_url.split('/')[1]
+    folder_route = current_url.split('/')[2..-1].join('/')
+    user = User.find_by(username: username)
+
+    if @current_folder = Folder.find_by(route: folder_route, user_id: user.id)
+      session[:folder_id] = @current_folder.id
+    else
+      render :file => "#{Rails.root}/public/404.html",  :status => 404
+    end
+  end
+
+  def index
+  end
+
+  def new
+    @folder = Folder.new
+  end
+
+  def create
+    parent_folder = Folder.find(session[:folder_id])
```

Comment:
@case-eee , how do you feel about storing information in the session as we are? We realized that we could also probably use sessions to store the user information when creating their account instead of using the hidden fields as we were. Is that how sessions are intended to be used?